Add Differential Expression module view

helenmiller16 opened the pull request:
<!-- IF THIS INVOLVES AUTHENTICATION: DO NOT SHARE YOUR USERNAME/PASSWORD, OR API KEYS/TOKENS IN THIS ISSUE - MOST LIKELY THE MAINTAINER WILL HAVE THEIR OWN EQUIVALENT KEY -->


<!--- Provide a general summary of your changes in the Title above -->

<!-- Before submitting your PR, make sure your code follows the guidelines 
outlined in the contribution guidelines and review the PR checklist below. 
In particular, make sure you follow the style guide for any coding languages 
you have edited, and add relevant tests and documentation. -->

## Description

1. Create ETL to run differential expression 
2. Create view to display table with differential expression results (only available at individual studies level)

This removes all dependencies on UpdateAnno package from LabKeyModules

## Related PRs

* https://github.com/RGLab/UITesting/pull/56
* https://github.com/RGLab/HIPCMatrix/pull/14

<!--- Link to any related PRs in UITesting, ImmuneSpaceR, etc. If this PR
makes any changes to UI functionality, there should be a related UITesting PR. -->

## Notion Ticket

https://www.notion.so/rglab/Differential-Expression-module-view-ade3f2fe65fe4a1fb08bc15e50f578be

## PR Checklist for reviewer

- [ ] Build it on your own local instance and test new features or changes.  
- [ ] Pass UITesting on your local instance on the matching UITesting branch.  
- [ ] It is functional and meets the requirements?   
- [ ] Are any shorthand variable names are included in the [Glossary](https://www.notion.so/rglab/Glossary-of-variable-abbreviations-e205838b1f534abc903fa8c2228a6d7f)?  
- [ ] Does it include descriptive code comments?   
- [ ] Does it follow the [Style Guide](https://github.com/RGLab/LabKeyModules/#style-guide) for any languages included?  
- [ ] Is it well-documented?
    * Module-specific README lists out any dependencies  
    * Sufficient instructions for setup  
    * Links to any relevant LabKey documentation   
- [ ] Are any new features tested via UITesting or jest?  
- [ ] Does it have any new dependencies? Is it justifiable?  
- [ ] Is there code duplication that could be reduced?  
- [ ] Are there performance improvements that could be made?

juyeongkim commented:
@helenmiller16 merge into `dev` please

juyeongkim commented on ActiveModules/web/ActiveModules.js:
```diff
@@ -42,10 +42,12 @@ LABKEY.ext.ActiveModules = Ext.extend(Ext.Panel, {
               ];
               if (!LABKEY.ext.ISCore.isStudyFolder) {
                 modules.pop();
+                modules.pop();
```
Why pop twice?

helenmiller16 replied:
Remove both data explorer (It's under "analyze" tab instead) and Differential Expression (only meant to be viewed at study level at the moment).

juyeongkim commented on DifferentialExpressionAnalysis/reports/schemas/DifferentialExpressionAnalysis.Rmd:
````diff
@@ -5,27 +5,22 @@ opts_chunk$set(echo = FALSE, message = FALSE)
 ```{r libraries, message = FALSE, echo = FALSE, warning = FALSE}
-library(UpdateAnno)
+library(HIPCMatrix)
 library(DT)
````
DT is not being used anymore.

helenmiller16 replied:
This report is not in use anymore so I deleted it. Module view is run from views/begin.html instead of R report. Which is more consistent with how React modules work with the frontend separate from R/data work.

juyeongkim commented on DifferentialExpressionAnalysis/reports/schemas/DifferentialExpressionAnalysis.report.xml:
```diff
@@ -1,6 +1,6 @@
-<ReportDescriptor>
-    <hidden>false</hidden>
+<ReportDescriptor reportName="run_DEA" >
+    <hidden>true</hidden>
```
👍

juyeongkim commented on DifferentialExpressionAnalysis/views/begin.html:
```diff
@@ -0,0 +1,29 @@
+         queryName: 'gene_expression_analysis_results'
+     });
+     de_webpart.render();
+     </script>
```
Remove tab

helenmiller16 commented:
I deleted DifferentialExpressionAnalysis.Rmd because the module view is now controlled from view/begin.html, which is more consistent with how the React modules work, with frontend code separated from R/data-related code. Also updated the table in the view to use DGEA_filteredGEAR query, which filters out NA gene names

juyeongkim reviewed (approved):
Thanks for addressing my comments. It looks good. 💯
